packages/syft/src/syft/oblv/model.py

```diff
@@ -0,0 +1,441 @@
+                        "--pcr2",
+                        depl.pcr_codes[2],
+                        "--port",
+                        "443",
+                        "--lport",
+                        str(connection_port),
+                        "--disable-pcr-check",
+                    ],
+                    stdout=log_file,
+                    stderr=log_file,
+                )
+            else:
+                process = subprocess.Popen(
+                    [
+                        "oblv",
+                        "connect",
+                        "--private-key",
+                        private_file_name,
+                        "--public-key",
+                        public_file_name,
+                        "--url",
+                        depl.instance.service_url,
+                        "--pcr0",
+                        depl.pcr_codes[0],
+                        "--pcr1",
+                        depl.pcr_codes[1],
+                        "--pcr2",
+                        depl.pcr_codes[2],
+                        "--port",
+                        "443",
+                        "--lport",
+                        str(connection_port),
+                    ],
+                    stdout=log_file,
+                    stderr=log_file,
+                )
+            log_file_read = open(log_file_name, "r")
+            while True:
+                d = log_file_read.readline()
+                if d.__contains__("Error:  Invalid PCR Values"):
+                    raise Exception("PCR Validation Failed")
+                if d.__contains__("Only one usage of each socket address"):
+                    raise Exception(
+                        "Another oblv proxy instance running. Either close that connection"
+                        + "or change the *connection_port*"
+                    )
+                elif d.lower().__contains__("error"):
+                    raise Exception(d)
+                elif d.__contains__("listening on"):
+                    break
+        except Exception as e:
+            print("Could not connect to Proxy")
+            raise e
+        else:
+            print(
+                "Successfully connected to proxy on port {}. The logs can be found at {}".format(
+                    connection_port, log_file_name
+                )
+            )
+        self.__conn_string = "http://127.0.0.1:" + str(connection_port)
+        self.__logs = log_file_name
+        self.__process = process
+        return
+
+    def get_uploaded_datasets(self):
+        if len(self.client) == 0:
+            raise Exception(
+                "No Domain Clients added. Set the propert *client* with the list of your domain logins"
+            )
+        if self.__conn_string is None:
+            raise Exception(
+                "proxy not running. Use the method connect_oblv_proxy to start the proxy."
+            )
+        elif self.__conn_string == "":
```

Comment:
NITPICK: `not self.__con_string`

Reply:
Solved by [2a66c5b](https://github.com/OpenMined/PySyft/pull/7215/commits/2a66c5b1e28ab894928014e38b9d8ed8378e8f5f)

Reply:
[75bbe29](https://github.com/OpenMined/PySyft/pull/7215/commits/75bbe295f140914eb8fb0c8c15b249f94e75c40d)